src/codegen.cpp

```diff
@@ -2510,7 +2519,10 @@ static bool emit_builtin_call(jl_cgval_t *ret, jl_value_t *f, jl_value_t **args,
             if (jl_is_leaf_type(tp0)) {
                 emit_expr(args[1], ctx);
                 assert(jl_is_datatype(tp0));
-                *ret = mark_julia_type(ConstantInt::get(T_size, jl_datatype_nfields(tp0)), false, jl_long_type, ctx);
+                jl_value_t *c = jl_box_long(jl_datatype_nfields(tp0));
+                if (ctx->linfo->def) // toplevel exprs are already rooted
+                    jl_add_linfo_root(ctx->linfo, c);
```

Comment:
`ctx->linfo->def` is `NULL` when running `test/random` (executing the code [at line 2585](https://github.com/JuliaLang/julia/blob/0db498cda3c72184a0484e045cb699b73c9b8c22/src/codegen.cpp#L2584) actually), so I copied [this similar check](https://github.com/JuliaLang/julia/blob/0db498cda3c72184a0484e045cb699b73c9b8c22/src/codegen.cpp#L2584) to guard against `jl_add_linfo_root` segfaulting. Where do I root this value without a valid `jl_method_t`?